skpro/regression/gradient_boosting/_ngboost.py

```diff
@@ -0,0 +1,342 @@
+        """Fit regressor to training data.
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        X : pandas DataFrame
+            feature instances to fit regressor to
+        y : pandas DataFrame, must be same length as X
+            labels to fit regressor to
+
+        Returns
+        -------
+        self : reference to self
+        """
+        from ngboost import NGBRegressor
+        from ngboost.scores import LogScore
+        from sklearn.tree import DecisionTreeRegressor
+
+        if self.estimator is None:
+            self.estimator = DecisionTreeRegressor(
+                criterion="friedman_mse",
+                min_samples_split=2,
+                min_samples_leaf=1,
+                min_weight_fraction_leaf=0.0,
+                max_depth=3,
+                splitter="best",
+                random_state=None,
+            )
+
+        dist_ngboost = self.dist_to_ngboost_instance()
+
+        # Score argument for NGBRegressor
+        ngboost_score = {
+            "LogScore": LogScore,
+        }
+        score = None
+        if self.score in ngboost_score:
+            score = ngboost_score[self.score]
+
+        self.ngb = NGBRegressor(
+            Dist=dist_ngboost,
+            Score=score,
+            Base=self.estimator,
+            natural_gradient=True,
+            n_estimators=self.n_estimators,
+            learning_rate=self.learning_rate,
+            minibatch_frac=self.minibatch_frac,
+            col_sample=self.col_sample,
+            verbose=self.verbose,
+            verbose_eval=self.verbose_eval,
+            tol=self.tol,
+            random_state=self.random_state,
+            validation_fraction=self.validation_fraction,
+            early_stopping_rounds=self.early_stopping_rounds,
+        )
+        self.ngb.fit(X, y)
+        return self
+
+    def _predict(self, X):
+        return self.ngb.predict(X)
+
+    def _pred_dist(self, X):
+        return self.ngb.pred_dist(X)
+
+    def _ngb_dist_to_skpro(self, **kwargs):
+        """
+        Convert NGBoost distribution object to skpro BaseDistribution object.
+
+        Parameters
+        ----------
+        pred_mean, pred_std and index and columns.
+
+        Returns
+        -------
+        skpro_dist (skpro.distributions.BaseDistribution):
+        Converted skpro distribution object.
+        """
+        from skpro.distributions.laplace import Laplace
+        from skpro.distributions.lognormal import LogNormal
+        from skpro.distributions.normal import Normal
+        from skpro.distributions.poisson import Poisson
+        from skpro.distributions.t import TDistribution
+
+        ngboost_dists = {
+            "Normal": Normal,
+            "Laplace": Laplace,
+            "TDistribution": TDistribution,
+            "Poisson": Poisson,
+            "LogNormal": LogNormal,
+        }
+
+        kwargs["index"] = kwargs["mu"].index
```

Comment:
I think this should be from `X.index` (from `predict_proba`), and `y.index` (from `fit`), not the index that ngboost returns.

Reply:
Yes that makes sense. I have now made it `X.index` inplace of `kwargs["mu"].index` however I am still unable to remove the 
```
                kwargs[skp_param] = self._check_y(y=kwargs[skp_param])
                # returns a tuple so taking only first index of the tuple
                kwargs[skp_param] = kwargs[skp_param][0]
```
As it is causing a shape mismatch error upon removing it.
`ValueError: shape mismatch: objects cannot be broadcast to a single shape.  Mismatch is between arg 0 with shape (111,) and arg 3 with shape (10,)`.
Where 111 expected Index but arg3 would be the number of columns which seems to be 10.

Reply:
do you have an explanation why this happens?